Index: src/main/java/org/apache/myfaces/component/html/ext/HtmlDataTable.java
===================================================================
--- src/main/java/org/apache/myfaces/component/html/ext/HtmlDataTable.java	(revision 558729)
+++ src/main/java/org/apache/myfaces/component/html/ext/HtmlDataTable.java	(working copy)
@@ -785,8 +785,10 @@
         }
         values[3] = _preserveSort;
         values[4] = _forceIdIndexFormula;
+        _sortColumn = getSortColumn();
+        _sortAscending = Boolean.valueOf(isSortAscending());        
         values[5] = _sortColumn;
-        values[6] = _sortAscending;
+        values[6] = _sortAscending;        
         values[7] = _sortProperty;
         values[8] = _sortable;
         values[9] = _renderedIfEmpty;
