Index: src/main/java/org/apache/myfaces/custom/datascroller/HtmlDataScrollerRenderer.java
===================================================================
--- src/main/java/org/apache/myfaces/custom/datascroller/HtmlDataScrollerRenderer.java	(revision 472630)
+++ src/main/java/org/apache/myfaces/custom/datascroller/HtmlDataScrollerRenderer.java	(working copy)
@@ -25,11 +25,11 @@
 import javax.faces.application.Application;
 import javax.faces.component.UIComponent;
 import javax.faces.component.UIParameter;
-import javax.faces.component.html.HtmlCommandLink;
 import javax.faces.component.html.HtmlOutputText;
 import javax.faces.context.FacesContext;
 import javax.faces.context.ResponseWriter;
 
+import org.apache.myfaces.component.html.ext.HtmlCommandLink;
 import org.apache.myfaces.shared_tomahawk.renderkit.RendererUtils;
 import org.apache.myfaces.shared_tomahawk.renderkit.JSFAttr;
 import org.apache.myfaces.shared_tomahawk.renderkit.html.HtmlRenderer;
