Index: trinidad/trinidad-impl/src/main/java/org/apache/myfaces/trinidadinternal/style/util/CSSGenerationUtils.java
===================================================================
--- trinidad/trinidad-impl/src/main/java/org/apache/myfaces/trinidadinternal/style/util/CSSGenerationUtils.java	(revision 553329)
+++ trinidad/trinidad-impl/src/main/java/org/apache/myfaces/trinidadinternal/style/util/CSSGenerationUtils.java	(working copy)
@@ -762,7 +762,36 @@
     else
     {
       // there are no namespaces in this selector.
-      mappedSelector = _convertPseudoClassesInSelector(selector);
+    	
+      //Separate pseudoclasses and then call it
+      int start = 0;
+      StringBuilder b = new StringBuilder();
+      for(int i = 0; i < selector.length(); i++){
+    	  char c = selector.charAt(i); 
+    	  if (c == ' ' ){
+    		  if (start == i){
+    			  //group of spaces
+    			  start = i+1; //Skip space    			      			  
+    		  }else{
+    			  String subSelector = selector.substring(start,i);
+    			  subSelector = _convertPseudoClassesInSelector(subSelector);
+    			  start = i+1; //Skip space
+    			  b.append(subSelector);
+    			  b.append(' ');
+    		  }
+    	  }
+      }
+      if (start == 0){
+    	  //there is no pseudoclasses in selector    	  
+    	  mappedSelector = _convertPseudoClassesInSelector(selector);
+      }else{
+    	  String subSelector = selector.substring(start);
+    	  subSelector = _convertPseudoClassesInSelector(subSelector);
+    	  b.append(subSelector);
+    	  mappedSelector = b.toString();
+      }
+      //previous line
+      //mappedSelector = _convertPseudoClassesInSelector(selector);
     }
     return mappedSelector;
   }
